File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        ls=[]
        if root:
            ls=self.inorderTraversal(root.left)
            ls.append(root.val)
            ls+=self.inorderTraversal(root.right)
        
        return ls
        

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> lstVals= new ArrayList<>();
        if(root==null) return lstVals;
        inorderTraversal(root,lstVals);
        return lstVals;
    }
  private void  inorderTraversal(TreeNode root,List<Integer> lstVals){
    if(root==null) return;
    
    
    inorderTraversal(root.left,lstVals);
      lstVals.add(root.val);
    inorderTraversal(root.right,lstVals);
    
}
}